lib/iris/experimental/um.py

```diff
@@ -520,7 +526,14 @@ def constants(name, dtype):
                     else:
                         offset = running_offset
                         data_provider = data_class(source, offset, word_size)
-                    klass = _FIELD_CLASSES[raw_headers[Field.LBREL_OFFSET]]
+                    lbrel_value = raw_headers[Field.LBREL_OFFSET]
+                    if lbrel_value == mdi_value:
```

Comment:
I've removed the changes I was putting in to handle both, and I'm now pulling mdi_value from self.integer_constants[21].

:)